File: go/hamming/hamming.go
package hamming

import "errors"

func Distance(a, b string) (count int, err error) {

    for i := 0; i < len(a); i++ {
        if (len(a) != len(b)) || (len(a) == 0)  || (len(b) == 0){
            return 0, errors.New("strings of unequal length")
        }
        if a[i] != b[i]{
            count++
        }
    }

    return
}

File: cpp/hamming/hamming.cpp
#include "hamming.h"

int hamming::compute(string a, string b){
    if (a.length() != b.length()){
		throw std::domain_error("Unequal size");
	}

	int dist = 0;
	for(int i=0; i<a.length(); i++){
		if (a[i] != b[i]){
			dist = dist + 1;
		}
	}
	return dist;
}
